java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/cp/CPMapTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.simulator.tests.cp;
+
+import com.hazelcast.cp.CPMap;
+import com.hazelcast.simulator.hz.HazelcastTest;
+import com.hazelcast.simulator.test.BaseThreadState;
+import com.hazelcast.simulator.test.annotations.Prepare;
+import com.hazelcast.simulator.test.annotations.Setup;
+import com.hazelcast.simulator.test.annotations.TimeStep;
+import com.hazelcast.simulator.utils.GeneratorUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CPMapTest extends HazelcastTest {
+    // number of cp groups to host the created maps; if (distinctMaps % maps) == 0 then there's a uniform distribution of maps
+    // over cp groups, otherwise maps are allocated per-cp group in a RR-fashion.
+    public int cpGroups = 1;
+    // number of distinct maps to create and use during the tests
+    public int maps = 1;
+    // number of distinct keys to create and use per-map; key domain is [0, keys)
+    public int keys = 1;
+    // size in bytes for each key's associated value
+    public int valueSizeBytes = 100;
+    private List<CPMap<Integer, String>> mapReferences;
+
+    private String v; // this is always the value associated with any key; exception is remove and delete
+
+
+    @Setup
+    public void setup() {
+        v = GeneratorUtils.generateAsciiString(valueSizeBytes);
+
+        // (1) create the cp groups names that will host the maps
+        String[] cpGroupNames = new String[cpGroups];
+        for (int i = 0; i < cpGroups; i++) {
+            cpGroupNames[i] = "cpgroup-" + i;
+        }
+
+        // (2) create the map names + associated proxies (maps aren't created until you actually interface with them)
+        mapReferences = new ArrayList<>();
```

Comment:
Can there be an option to use the default cpGroup just like the IAtomicReferenceTest has?

Reply:
Will add it - the `default` CP Group doesn't actually exist either until you create a CP object that either is not fully qualified, e.g. `mymap`, or explicitly qualified, e.g. `mymap@default`. So, in that respect it's no different.